### ISSUE TITLE
SPIKE: Test the performance of an additional replica

### ISSUE THREAD

**Issue by mmarcotte** [ustaxcourt/ef-cms#1747]
So, as of right now we use 1 shard and 1 replica per index. However, we have three nodes. That means that each index's data is stored on two out of the three servers. It's been working fine so far, but what happens when we add another replica?

That means that each server will have a copy of each index. That must also mean that we will end up using more disk space. It will also mean that three servers can handle the traffic that previously only two could handle. If we scale horizontally, it will also mean that we will likely want to replicas, too.

**Comment by mmarcotte** [comment]
Good news, we added a replica to our cluster and observed about a 25% performance increase (using CPU as a measurement) from the addition in a like-for-like order search test. 

More good news is that the [recent change](https://github.com/ustaxcourt/ef-cms/issues/1731) to `r5.xlarge.elasticsearch` due to JVM Memory Pressure showed significant improvements in search performance. Feeling more confident than ever in our cluster's health and ability to withstand increased traffic.

The test was 5 minutes of concurrent searching, with an arrivalRate of 40 (that's 40 searches per second). 

1 replica: spiked to 50%
2 replicas: spiked to 36%

![1756-documentation](https://user-images.githubusercontent.com/5023502/138333926-c2a5439c-bbec-42ef-936d-79b64ac505f6.png)
 
We continued to test with an arrival rate of 80, and the cluster continued to withstand the additional load.

An added benefit of having the additional cluster is that if two of our nodes were to go down, one would still have a copy of all of our information. And if one goes down, the other two still operate as we would today (with each index living on two nodes in our cluster).

The recommendation is to adjust our config to have 2 replica shards to distribute the data completely across all three of our nodes. Creating [a story](https://github.com/ustaxcourt/ef-cms/issues/1756) to do that.

It's clear that the slow-down limiter was also having an effect on our ability to measure the requests per second. That is currently being removed in favor of a hard cap of searches per second (and perhaps by minute). We'll conduct another test to get this threshold value before going live with External Search.
